scale-tests/deploy-dispatchers.py

```diff
@@ -67,84 +80,158 @@ def no_stdout():
+
+    # create drivers & executors role quotas
+    drivers_role = setup_role(service_name, "drivers", quota_options)
+    executors_role = setup_role(service_name, "executors", quota_options)
+
+    setup_spark_security(service_name, drivers_role, service_account_info)
+
+    service_options = scale_tests_utils.get_service_options(service_name, service_account_info, additional_options, config_path)
 
-            # install dispatcher with appropriate role
-            options["service"]["role"] = drivers_role
+    # create drivers & executors role quotas
+    drivers_role = setup_role(service_name, "drivers", quota_options)
+    executors_role = setup_role(service_name, "executors", quota_options)
 
-            sdk_install.install(
-                arguments['--package-name'],
-                service_name,
-                0,  # This library cannot see this non-SDK task.
-                additional_options=options,
-                wait_for_deployment=False)
+    # install dispatcher with appropriate role
+    service_options["service"]["role"] = drivers_role
 
-            outfile.write("{},{},{}\n".format(service_name, drivers_role, executors_role))
+    expected_task_count = service_task_count(service_options)
+    log.info("Expected task count: %s", expected_task_count)
+
+    log.info("Installing %s index %s as %s", package_name, index, service_name)
+
+    sdk_install.install(
+        package_name,
+        service_name,
+        expected_task_count,
+        additional_options=service_options,
+        wait_for_deployment=False,
+        insert_strict_options=False)
+
+    return {"package_name": package_name,
+            "roles": {"drivers": drivers_role, "executors": executors_role},
+            "service_account_info": service_account_info,
+            **service_options}
+
+
+def deploy_dispatchers(
+    num_dispatchers: int,
+    service_name_base: str,
+    output_file: str,
+    linux_user: str,
+    options: typing.Dict,
+    quota_options: typing.Dict):
+
+    shakedown.run_dcos_command("package install spark --cli --yes", raise_on_error=True)
+    with open(output_file, "w") as outfile:
+
+        dispatchers = []
+        for i in range(num_dispatchers):
+            dispatcher_settings = install_package("spark", service_name_base, i,
+                                                linux_user,
+                                                lambda x: 0,
+                                                None,
+                                                options,
+                                                quota_options)
+            dispatchers.append(dispatcher_settings)
+            outfile.write("{},{},{}\n".format(dispatcher_settings["service"]["name"],
+                                              dispatcher_settings["roles"]["drivers"],
+                                              dispatcher_settings["roles"]["executors"]))
```

Comment:
Maybe we'll need to write their service account and secret too, so that we have it handy for submitting jobs.